open_instruct/finetune.py

```diff
@@ -667,8 +720,24 @@ def main():
             active_dataloader = train_dataloader
         for step, batch in enumerate(active_dataloader):
             with accelerator.accumulate(model):
-                outputs = model(**batch, use_cache=False)                
-                loss = outputs.loss
+                outputs = model(**batch, use_cache=False)
+                if args.reduce_loss == 'mean':
+                    loss = outputs.loss
+                else:
+                    # we are going to calculate the loss ourself for sum avging.
```

Comment:
Probably it's better to give more explanation here, and refer to this discussion: https://github.com/huggingface/transformers/issues/24725?